### PR TITLE
Fixes a flake with the LoadReporter end2end test.

### PR DESCRIPTION
**Pull request by nicolasnoble**
I *believe* the test is wrong, based on the .proto description of the LoadReporter.

The protocol described in src/proto/grpc/lb/v1/load_reporter.proto has the ReportLoad rpc returns a stream of LoadReportResponse, which itself has a repeated field of Load messages. The comment before it states:

> It is not strictly necessary to aggregate all entries into one entry per <tag, user_id> tuple, although it is preferred to do so.

Debugging the issue shows we are in fact properly getting all 3 expected load report types, just in two separate messages instead of a single one.

This new test codepath will coalesce the load report responses, and also addresses the fact the original test wasn't verifying that we were getting the 3 expected types.

@ctiller: github's UI is suggesting you're the most appropriate reviewer for this, but feel free to reassign if that's not the case.

### CODE REVIEW

**Review by markdroth — COMMENTED** [review]
This looks reasonable to me.  I'm not 100% certain from the proto if this is the expected behavior or not, but it certainly seems reasonable.  @vishalpowar might be able to tell us more from the balancer side.

Even if we don't know for sure whether or not this is okay, I think it's probably fairly safe to assume it's fine, since the load reporting code hasn't changed in a very long time, so if there is a bug here, it's obviously not something that anyone has noticed.  Although to be fair, there may not be very many people actually using it, given how hard it is to use OpenCensus in OSS gRPC C++.

File: test/cpp/end2end/server_load_reporting_end2end_test.cc [hunk 1]
```diff
@@ -142,39 +142,50 @@ TEST_F(ServerLoadReportingEnd2endTest, BasicReport) {
   const std::string& lb_id = response.initial_response().load_balancer_id();
   gpr_log(GPR_INFO, "Initial response received (lb_id: %s).", lb_id.c_str());
   ClientMakeEchoCalls(lb_id, "LB_TAG", kOkMessage, 1);
-  while (true) {
+
+  unsigned loadCount = 0;
```
[the remark this answers is not in the document]
Comment (nicolasnoble) [reply test/cpp/end2end/server_load_reporting_end2end_test.cc]: Done.

File: test/cpp/end2end/server_load_reporting_end2end_test.cc [hunk 2]
```diff
@@ -142,39 +142,50 @@ TEST_F(ServerLoadReportingEnd2endTest, BasicReport) {
...
-                    1);
-          ASSERT_EQ(load.metric_data().Get(0).total_metric_value(),
-                    kMetricValue);
-        }
+    for (const auto& load : response.load()) {
+      loadCount++;
+      if (load.in_progress_report_case()) {
+        // The special load record that reports the number of in-progress
+        // calls.
+        ASSERT_EQ(load.num_calls_in_progress(), 1);
```
[the remark this answers is not in the document]
Comment (nicolasnoble) [reply test/cpp/end2end/server_load_reporting_end2end_test.cc]: Done.

**Review by markdroth — APPROVED** [review]
You might want to double check the intent with @vishalpowar before merging.  But this looks good to me.  Thanks for fixing this!

**Review by vishalpowar — APPROVED** [review]
LGTM, as @markdroth mentioned in the other comments. This code has not been touched for a long time and hence this should be a safe change. Thanks for cleaning this up.

**Pull request merged**